include/BSMPT/bounce_solution/bounce_solution.h

```diff
@@ -0,0 +1,557 @@
+                 CoexPhases &phase_pair_in,
+                 double UserDefined_vwall_in,
+                 double UserDefined_epsturb_in,
+                 int MaxPathIntegrations_in);
+
+  /**
+   * @brief Construct a new Bounce Sol Calc object. This class takes as input a
+   * pair of coexisting phases.
+   *
+   * @param pointer_in model pointer
+   * @param MinTracerIn minimum tracer pointer
+   * @param phase_pair_in pair of coexisting phases
+   * @param UserDefined_vwall_in is the input value for v_wall. If = -1$ then we
+   * use the approximation coming from https://arxiv.org/abs/2210.16305. If =
+   * -2$ then we use the upper bound from https://arxiv.org/abs/2305.02357
+   * @param UserDefined_epsturb_in is the input value for epsturb. If [0..1] set
+   * to value, for -1 we use the upper bound from
+   * https://arxiv.org/abs/1704.05871
+   * @param MaxPathIntegrations_in max number of path integrations
+   * @param GroupElements_In List of allowed potential symmetries
+   */
+  BounceSolution(std::shared_ptr<Class_Potential_Origin> &pointer_in,
+                 std::shared_ptr<MinimumTracer> &MinTracer_in,
+                 CoexPhases &phase_pair_in,
+                 double UserDefined_vwall_in,
+                 double UserDefined_epsturb_in,
+                 int MaxPathIntegrations_in,
+                 std::vector<Eigen::MatrixXd> GroupElements_in);
+
+  /**
+   * @brief Initially we have no idea where the transition can occur, therefore
+   * we scan the complete temperature range
+   */
+  void GWInitialScan();
+
+  /**
+   * @brief Calculate the euclidian action of the transition from false to true
+   * phase of phase pair.
+   *
+   * @param T temperature
+   * @param smart
+   */
+
+  void CalculateActionAt(double T, bool smart = true);
+
+  /**
+   * @brief If solution were found by the GWInitialScan() then we scan
+   * temperature range in the vicinity such that we are get a enough sample to
+   * then do the extrapolation.
+   *
+   */
+  void GWSecondaryScan();
+
+  /**
+   * @brief Do linear extrapolations to calculate action at higher temperatures
+   *
+   */
+  void GWScanTowardsHighAction();
+
+  /**
+   * @brief Do linear extrapolations to calculate action at lower temperatures
+   *
+   */
+  void GWScanTowardsLowAction();
+
+  /**
+   * @brief Set the Bounce Sol object
+   *
+   */
+  void SetBounceSol();
+
+  /**
+   * @brief Get the bubble wall velocity
+   * @return vb
+   */
+  double GetWallVelocity();
+
+  /**
+   * @brief Get epsturb
+   * @return epsturb
+   */
+  double GetEpsTurb();
+
+  /**
+   * @brief SetGstar Set gstar
+   */
+  void SetGstar(const double &gstar_in);
+
+  /**
+   * @brief GetGstar Get gstar
+   */
+  double GetGstar();
+
+  /**
+   * @brief SetCriticalTemp Set critical temperature
+   */
+  void SetCriticalTemp(const double &T_in);
+
+  /**
+   * @brief GetCriticalTemp Get critical temperature
+   */
+  double GetCriticalTemp();
```

Comment:
```suggestion
  double GetCriticalTemp() const;
```

All the getter functions should be possible to make const